src/provider.test.tsx

```diff
@@ -281,4 +281,34 @@ describe('LDProvider', () => {
     expect(mockLDClient.on).toHaveBeenCalledWith('change', expect.any(Function));
     expect(newState).toEqual({ flags: { 'another-test-flag': false, 'test-flag': false } });
   });
+
+  test(`if props.deferInitialization is true, ld client will only initialize once props.user is defined`, async () => {
+    const options: LDOptions = { bootstrap: {} };
+    const props: ProviderConfig = { clientSideID, deferInitialization: true, options };
+    const LaunchDarklyApp = (
+      <LDProvider {...props}>
+        <App />
+      </LDProvider>
+    );
+    const renderer = create(LaunchDarklyApp);
+    const instance = renderer.root.findByType(LDProvider).instance as EnhancedComponent;
+
+    await instance.componentDidMount();
+
+    expect(mockInitLDClient).toHaveBeenCalledTimes(0);
+
+    const user: LDUser = { key: 'yus', name: 'yus ng' };
+    const newProps = { ...props, user };
+    const UpdatedLaunchDarklyApp = (
+      <LDProvider {...newProps}>
+        <App />
+      </LDProvider>
+    );
+    renderer.update(UpdatedLaunchDarklyApp);
+    if (instance.componentDidUpdate) {
+      await instance.componentDidUpdate(props, {});
```

Comment:
Linting failed here with the following errors:

```bash
ERROR: /src/provider.test.tsx:309:7 - Invalid 'await' of a non-Promise value.
ERROR: /src/provider.test.tsx:309:13 - Expression has type `void`. Put it on its own line as a statement.
```

You should be able to reproduce by running `yarn lint:all`.

Reply:
ahh, good catch. I have to update the types so that `componentDidUpdate` returns a `Promise`, similar to `componentDidMount`.